classes/Attachment.php

```diff
@@ -173,13 +173,24 @@ public static function deleteProductAttachments($idProduct)
      * @return bool true if success
      */
     public function attachProduct($idProduct)
+    {
+        return static::associateProductAttachment((int) $idProduct, (int) $this->id);
+    }
+
+    /**
+     * @param int $productId
+     * @param int $attachmentId
+     *
+     * @return bool true if success
+     */
+    public static function associateProductAttachment(int $productId, int $attachmentId): bool
     {
         $res = Db::getInstance()->execute('
 			INSERT INTO ' . _DB_PREFIX_ . 'product_attachment
 				(id_attachment, id_product) VALUES
-				(' . (int) $this->id . ', ' . (int) $idProduct . ')');
+				(' . $attachmentId . ', ' . $productId . ')');
```

Comment:
I think there is no `declare strict types` in this class (because adding it would be a BC break).

So is `(int $productId, int $attachmentId): bool` enough to ensure these are integers and consequently secure the query against SQL injection ?

Reply:
A string will be converted to integer. (If it is a text php will throw TypeError). So I suppose it is enough